From 50dd511f73c22341384c355105ebbd775d986fff Mon Sep 17 00:00:00 2001
From: Cheng-mean Liu <soccerl@microsoft.com>
Date: Tue, 11 Jul 2017 16:50:36 -0700
Subject: [PATCH 19/21] Added vsock transport support to 9pfs

Signed-off-by: Cheng-mean Liu <soccerl@microsoft.com>
Origin: https://github.com/Microsoft/opengcs/blob/master/kernelconfig/4.11/patch_9pfs_vsock-transport.patch
---
 net/9p/trans_fd.c | 85 ++++++++++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 84 insertions(+), 1 deletion(-)

diff --git a/net/9p/trans_fd.c b/net/9p/trans_fd.c
index 7bc2208b6cc4..dc34352145df 100644
--- a/net/9p/trans_fd.c
+++ b/net/9p/trans_fd.c
@@ -44,8 +44,9 @@
 #include <net/9p/9p.h>
 #include <net/9p/client.h>
 #include <net/9p/transport.h>
-
 #include <linux/syscalls.h> /* killme */
+//#include <linux/kgdb.h>
+#include <linux/vm_sockets.h>
 
 #define P9_PORT 564
 #define MAX_SOCK_BUF (64*1024)
@@ -155,6 +156,7 @@ struct p9_trans_fd {
 	struct p9_conn conn;
 };
 
+
 static void p9_poll_workfn(struct work_struct *work);
 
 static DEFINE_SPINLOCK(p9_poll_lock);
@@ -740,6 +742,7 @@ static int parse_opts(char *params, struct p9_fd_opts *opts)
 	opts->wfd = ~0;
 	opts->privport = 0;
 
+
 	if (!params)
 		return 0;
 
@@ -1035,6 +1038,72 @@ p9_fd_create(struct p9_client *client, const char *addr, char *args)
 	return 0;
 }
 
+static int
+p9_fd_create_vsock(struct p9_client *client, const char *addr, char *args)
+{
+	int err;
+	struct socket *csocket;
+	struct sockaddr_vm server_socket_addr;
+	struct p9_fd_opts opts;
+
+	err = parse_opts(args, &opts);
+	if (err < 0)
+		return err;
+
+	csocket = NULL;
+
+	// for debugging purpose only
+        pr_err("%s:%s\n", __func__, addr);
+	// kgdb_breakpoint();
+
+        // create socket
+	err = __sock_create(current->nsproxy->net_ns,
+		                AF_VSOCK,
+		                SOCK_STREAM,
+                                0,
+                                &csocket, 1);
+	if (err) {
+		pr_err("%s:__sock_create (%d): problem creating socket (err=%d)\n",
+			__func__, task_pid_nr(current), err);
+		return err;
+	}
+
+	// server socket address information
+	memset((char *)&server_socket_addr, 0, sizeof(struct sockaddr_vm));
+	server_socket_addr.svm_family = AF_VSOCK;
+	server_socket_addr.svm_reserved1 = 0;
+        server_socket_addr.svm_cid = VMADDR_CID_HOST;
+
+        /* Connecting to the host's 0000pppp-facb-11e6-bd58-64006a7986d3 */ 
+	server_socket_addr.svm_port = opts.port;
+
+        pr_err("%s:opts.port=(%d)(0x%x)\n", __func__, opts.port, opts.port);
+        pr_err("%s: service_id:(hex) 0000%x%x-facb-11e6-bd58-64006a7986d3\n",
+                __func__, 
+                (__u8)((opts.port & 0xff00) >> 8),
+                (__u8)(opts.port & 0x00ff));
+
+        pr_err("%s: connecting", __func__);
+	err = csocket->ops->connect(csocket,
+		                    (struct sockaddr *)&server_socket_addr,
+		                    sizeof(struct sockaddr_vm), 0);
+	if (err < 0) {
+		pr_err("%s:connect (%d): problem connecting socket to %s (err = %d)\n",
+			__func__, task_pid_nr(current), addr, err);
+		sock_release(csocket);
+		return err;
+	}
+
+        pr_err("%s: open socket", __func__);
+        err = p9_socket_open(client, csocket);
+        if (err < 0) {
+            pr_err("%s: p9_socket_open failed\n", __func__);
+         }
+
+        pr_err("Leaving %s\n", __func__);
+	return err;
+}
+
 static struct p9_trans_module p9_tcp_trans = {
 	.name = "tcp",
 	.maxsize = MAX_SOCK_BUF,
@@ -1071,6 +1140,18 @@ static struct p9_trans_module p9_fd_trans = {
 	.owner = THIS_MODULE,
 };
 
+static struct p9_trans_module p9_vsock_trans = {
+	.name = "vsock",
+	.maxsize = MAX_SOCK_BUF,
+	.def = 0,
+	.create = p9_fd_create_vsock,
+	.close = p9_fd_close,
+	.request = p9_fd_request,
+	.cancel = p9_fd_cancel,
+	.cancelled = p9_fd_cancelled,
+	.owner = THIS_MODULE,
+};
+
 /**
  * p9_poll_proc - poll worker thread
  * @a: thread state and arguments
@@ -1108,6 +1189,7 @@ int p9_trans_fd_init(void)
 	v9fs_register_trans(&p9_tcp_trans);
 	v9fs_register_trans(&p9_unix_trans);
 	v9fs_register_trans(&p9_fd_trans);
+	v9fs_register_trans(&p9_vsock_trans);
 
 	return 0;
 }
@@ -1118,4 +1200,5 @@ void p9_trans_fd_exit(void)
 	v9fs_unregister_trans(&p9_tcp_trans);
 	v9fs_unregister_trans(&p9_unix_trans);
 	v9fs_unregister_trans(&p9_fd_trans);
+	v9fs_unregister_trans(&p9_vsock_trans);
 }
-- 
2.11.1

