From 48efc524be9a3f19f515111ef17bd8d2b384c13c Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 8 Jul 2011 20:25:16 +0200
Subject: [PATCH 407/437] Add localversion for -RT release

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 localversion-rt | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 localversion-rt

diff --git a/localversion-rt b/localversion-rt
new file mode 100644
index 000000000000..90290c642ed5
--- /dev/null
+++ b/localversion-rt
@@ -0,0 +1 @@
+-rt29
-- 
2.19.1

