From 422b603925cb4488f30f11104413d82e924698ff Mon Sep 17 00:00:00 2001
From: Tiejun Chen <tiejunc@vmware.com>
Date: Sun, 26 Aug 2018 17:03:10 -0700
Subject: [PATCH 1/1] Miss the irq_regs.h file

Signed-off-by: Tiejun Chen <tiejunc@vmware.com>
---
 drivers/hv/vmbus_drv.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/hv/vmbus_drv.c b/drivers/hv/vmbus_drv.c
index 4970f10b..ba41140a 100644
--- a/drivers/hv/vmbus_drv.c
+++ b/drivers/hv/vmbus_drv.c
@@ -47,6 +47,8 @@
 #include <linux/random.h>
 #include "hyperv_vmbus.h"
 
+#include <asm/irq_regs.h>
+
 struct vmbus_dynid {
 	struct list_head node;
 	struct hv_vmbus_device_id id;
-- 
2.17.1

