From 10a87e7fb2bb7bf20af50a4fa11e60ead439e2f5 Mon Sep 17 00:00:00 2001
From: Steven Rostedt <rostedt@goodmis.org>
Date: Mon, 14 May 2018 12:48:00 -0400
Subject: [PATCH 414/414] Linux 4.14.40-rt30 REBASE

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 90290c642ed5..b72862e06be4 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt29
+-rt30
-- 
2.17.0

