From 9ff100f0120bbcb8e5209ee9a3820a421372d257 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Fri, 4 Jan 2019 20:59:15 -0500
Subject: [PATCH 450/450] Linux 4.14.87-rt50 REBASE

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 90290c642ed5..42c384668389 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt29
+-rt50
-- 
2.19.2

