From aad78d70e503de59191ef6b3a9a86ef08857e902 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Tue, 31 Mar 2020 12:32:04 -0400
Subject: [PATCH 328/328] Linux 4.19.106-rt46 REBASE

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 1199ebade17b..272158183778 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt16
+-rt46
-- 
2.25.1

